ibtk/examples/VCViscousSolver/input2d

```diff
@@ -0,0 +1,163 @@
+u {
+   function_0 = "X_0^2*X_1^2"
+   function_1 = "X_0*X_1"
+}
+
+f {
+   function_0  = "1.0 + 2.0*X_0*X_1 + 2*X_0^2 + 4.0*X_1^2 + 4.0*X_0^3*X_1 + 8.0*X_0*X_1^3"
+   function_1  = "2.0*X_0^2 + X_1^2 + 4.0*X_0*X_1 + 6.0*X_0^2*X_1^2"
+}
+
+mu {
+   function = "1.0 + X_0*X_1"
+}
+
+// u velocity
+VelocityBcCoefs_0 {
+
+   u_fcn = "X_0^2*X_1^2"
+      
+   acoef_function_0 = "1.0"
+   acoef_function_1 = "1.0"
+   acoef_function_2 = "1.0"
+   acoef_function_3 = "1.0"
+
+   bcoef_function_0 = "0.0"
+   bcoef_function_1 = "0.0"
+   bcoef_function_2 = "0.0"
+   bcoef_function_3 = "0.0"
+
+   gcoef_function_0 = u_fcn
+   gcoef_function_1 = u_fcn
+   gcoef_function_2 = u_fcn
+   gcoef_function_3 = u_fcn
+   
+}
+
+// v velocity
+VelocityBcCoefs_1 {
+
+   v_fcn = "X_0*X_1"
+   
+   acoef_function_0 = "1.0"
+   acoef_function_1 = "1.0"
+   acoef_function_2 = "1.0"
+   acoef_function_3 = "1.0"
+
+   bcoef_function_0 = "0.0"
+   bcoef_function_1 = "0.0"
+   bcoef_function_2 = "0.0"
+   bcoef_function_3 = "0.0"
+
+   gcoef_function_0 = v_fcn
+   gcoef_function_1 = v_fcn
+   gcoef_function_2 = v_fcn
+   gcoef_function_3 = v_fcn
+
+}
+
+
+
+solver_type = "VC_VELOCITY_PETSC_KRYLOV_SOLVER"
+solver_db {
+   ksp_type          = "fgmres"
+   pc_type           = "shell"
+   rel_residual_tol  = 1.0e-12
+   abs_residual_tol  = 1.0e-15
+   initial_guess_nonzero = TRUE
+}
+
+precond_type = "VC_VELOCITY_PETSC_LEVEL_SOLVER"  //"VC_VELOCITY_POINT_RELAXATION_FAC_PRECONDITIONER"
```

Comment:
It is clear that there needs to be a specialization here, since this solver relies on the actual discretization.

Reply:
Even in KrylovSolver it needs matrix-free discretization, i.e. operator other than SCLaplaceOperator.